webapp/components/displayAmount.tsx

```diff
@@ -0,0 +1,66 @@
+import Big from 'big.js'
+import { TokenLogo } from 'components/tokenLogo'
+import { Tooltip } from 'components/tooltip'
+import { ComponentType, Fragment, ReactNode } from 'react'
+import { Token } from 'types/token'
+import { formatNumber } from 'utils/format'
+
+type CustomContainer = ComponentType<{
+  children?: ReactNode
+}>
+
+const DefaultTextContainer: CustomContainer = ({ children }) => (
+  <span>{children}</span>
+)
+
+type Props = {
+  amount: string
+  amountContainer?: CustomContainer
+  container?: CustomContainer
+  symbolContainer?: CustomContainer
+  showSymbol?: boolean
+} & ({ token: Token } | { symbol: Token['symbol'] })
+
+export const DisplayAmount = function ({
+  amountContainer: AmountContainer = DefaultTextContainer,
+  container: Container = Fragment,
+  amount,
+  showSymbol = true,
+  symbolContainer: SymbolContainer = DefaultTextContainer,
+  ...props
+}: Props) {
+  const formattedAmount = formatNumber(amount)
+
+  const symbol = 'symbol' in props ? props.symbol : props.token.symbol
+  const bigAmount = Big(amount)
+  const notZero = !bigAmount.eq(0)
+  // Only show dots for small numbers, less than the max 6 digits we're showing
+  // for formatted numbers.
+  const showDots = bigAmount.lt(0.000001) && notZero
```

Comment:
This is the same as the old "<0.00001" label we said we do not want to use because some degens may not understand math. We either re-evaluate using that for small amounts or use the elipsis when numbers less than 1 are removed its less-significant (rightmost) digits as we discussed previously. cc/ @dvnahuel

Reply:
As discussed with you and Nahuel, we will now only show decimals if the amount is less than 1, and we're truncating decimals.
Applied here [4765112](https://github.com/hemilabs/ui-monorepo/pull/898/commits/4765112f910695c0c69aaa17365deb05d973e741)